### DISCUSSION TITLE
Communication between two crazyflies

### DISCUSSION THREAD

**Discussion by FrancisGoncalves** [bitcraze/discussions#1039] [Q&A] [answered]
Hi! 

We are working on a project with a system composed of two crazyflies connected through a rigid bar. We are using a motion-capture system to estimate the position and attitude of the whole body. Our idea is to use only one of the microcontrollers to compute the control signals for both crazyflies. For this purpose, I have been reading about how to transmit data from one crazyflie to the other, and, from my understanding, P2P communication is probably the easiest way to go. However, we are using the radio to transmit motion-capture system data and I read that P2P communication does not work well when using the radio at the same time. Another thing we thought of is to connect both crazyflies with wires, either using the I2C or UART communication protocol. Again, reading other discussions, I understand that both crazyflies are set as masters in the I2C communication protocol, so maybe the UART would be the simplest way to go. I was wondering if you have any suggestions on how to achieve communication between the two crazyflies. In general, we would need high-frequency communication, around 500 Hz at least.

Thank you,
Francisco

**Comment by ataffanel** [comment, marked as answer]
I agree with your reasoning :-): Radio will be hard together with the positioning. I2C will be a problem because both Crazyflies want to be master. This should not be a problem, I2C can work in multi-master mode, but the stm32f405 I2C peripheral has a lot of bugs that makes that near impossible.

So the best option would indeed be the UART. This is especially good because you can just stream values over a uart, no need to do any extra packet or connection handling. Intuitively I would implement a very simple protocol with something like a sync bits: having frames of bytes where only the fist byte has a bit at a certain value, this way it is possible to very easily sync on the stream on the receiving side and it is something trivial to encode and send quickly on the sending side.

Assuming you have frames of 10 Bytes (4 16bits motors PWM + some sync housekeeping bytes), at 500Hz on a uart that means at least 40Kbaud. So at 115200 baud you can have a pretty wide margin.

**Reply by FrancisGoncalves** [reply]
Sounds good, thank you a lot for the answer!
